library/Translate/src/Juvix/FrontendContextualise/InfixPrecedence/Environment.hs

```diff
@@ -53,18 +92,71 @@ newtype Context a = Ctx {antiAlias :: ContextAlias a}
       HasSink "old" (Old Context.T),
       HasSource "old" (Old Context.T)
     )
-    via StateField "old" (ContextAlias)
+    via StateField "old" ContextAlias
   deriving
     ( HasState "new" (New Context.T),
       HasSink "new" (New Context.T),
       HasSource "new" (New Context.T)
     )
     via StateField "new" ContextAlias
+  deriving
+    ( HasReader "dispatch" EnvDispatch,
+      HasSource "dispatch" EnvDispatch
+    )
+    via ReaderField "dispatch" ContextAlias
   deriving
     (HasThrow "error" Error)
     via MonadError ContextAlias
 
+--------------------------------------------------------------------------------
+-- Generic Interface Definitions for Environment lookup
+--------------------------------------------------------------------------------
+
+class Query a where
+  type QueryConstraint a (m :: * -> *) :: Constraint
+  queryInfo' ::
+    (SymbLookup sym, QueryConstraint a m) => sym -> a -> m (Maybe [Context.Information])
+
+instance Query (SingleDispatch a b c) where
+  type QueryConstraint _ m = MinimalEnv a b c m
+  queryInfo' sym SingleDispatch = do
+    undefined
+
+instance Query EnvDispatch where
+  type QueryConstraint _ m = TransitionMap m
+  queryInfo' sym EnvDispatch = do
+    looked <- lookup sym
+    lookedO <- ask sym
+    let -- dealing with the second map winning lookup
+        -- Main rules:
```

Comment:
Is this explicitly documented somewhere (somewhere else)?

Reply:
also wondering if this function should be top-level

Reply:
This should be gone once we don't have to deal with two maps at once, It'll be promted when I have to reproduce the logic again in another env effect